internal/common/objects/IGameState.go

```diff
@@ -0,0 +1,13 @@
+package objects
+
+import (
+	"github.com/google/uuid"
+)
+
+/*
+IGameState is an interface for GameState that objects will use to get the current game state
+*/
+type IGameState interface {
+	GetLootBoxes() map[uuid.UUID]*LootBox
+	GetMegaBikes() map[uuid.UUID]*MegaBike
```

Comment:
Might be worth having the interface instead - prevents users from changing the MegaBike state outside of the server. Either that, or don't pass it by pointer. I think interface is cleaner.

The general idea for this data structure, I thought, was to allow copies of the game state to be accessed be agents. As such, I would only pass things by value, not by reference

Reply:
i.e., `GetMegaBikes() map[uuid.UUID]IMegaBike`